File: HJ_API/SIGESPROC.Common/Models/ViajesViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIGESPROC.Common.Models
{
    public class ViajesViewModel
    {
        public int viaj_id { get; set; }
        public int sucu_id { get; set; }
        public int user_user_id { get; set; }
        public int trans_id { get; set; }
        public DateTime viaj_fecha { get; set; }
        public decimal total_km { get; set; }
        public string sucu_nombre { get; set; }
        public decimal total_a_pagar { get; set; }

        public int? vide_id { get; set; }
        public int? cola_id { get; set; }
        public decimal? distancia_km { get; set; }
        public decimal? total_a_pagar_detalle { get; set; }
        public int? cosu_id { get; set; }
        public DateTime? fecha_viaje { get; set; }

        public string trans_nombre { get; set; }
        public string trans_apellido { get; set; }
        public decimal? trans_tarifa_por_km { get; set; }
        public bool? trans_activo { get; set; }

        public bool? cola_activo { get; set; }
        public decimal? distancia_sucursal { get; set; }

        public string cola_nombre { get; set; }
        public string cola_apelllido { get; set; }
        public string cola_sexo { get; set; }
        public string cola_email { get; set; }
        public string cola_direccion { get; set; }
        public bool? colaborador_activo { get; set; }

        public string user_username { get; set; }
        public string user_role { get; set; }

    }
}

File: HJ_Front/src/app/demo/models/modelsplanilla/viajesviewmodel.ts

export interface ViajeEncabezado {
    sucu_id: number; 
    user_user_id: number; 
    viaj_fecha: string; 
    trans_id: number; 
  }
  

  export interface ViajeDetalle {
    viaj_id: number; 
    cola_id: number; 
    distancia_km: number; 
    total_a_pagar: number; 
    cosu_id: number;
  }


  export interface ViajeConDetallesViewModel {
    viaj_id: number;
    sucu_id: number;
    user_user_id: number;
    sucu_nombre: string;
    trans_id: number;
    viaj_fecha: string;
    total_km: number;
    total_a_pagar: number;
    vide_id: number | null;
    cola_id: number | null;
    distancia_km: number | null;
    total_a_pagar_detalle: number | null;
    cosu_id: number | null;
    fecha_viaje: string | null; 
    trans_nombre: string;
    trans_apellido: string;
    trans_tarifa_por_km: number | null;
    trans_activo: boolean | null;
    cola_activo: boolean | null;
    distancia_sucursal: number | null;
    cola_nombre: string | null;
    cola_apelllido: string | null;
    cola_sexo: string | null;
    cola_email: string | null;
    cola_direccion: string | null;
    colaborador_activo: boolean | null;
    user_username: string;
    user_role: string;
  }
  
  